core/primitives/src/trie_key.rs

```diff
@@ -172,6 +195,17 @@ impl TrieKey {
                 res.extend(ACCOUNT_DATA_SEPARATOR);
                 res.extend(key);
             }
+            #[cfg(feature = "protocol_feature_transaction_hashes_in_state")]
+            TrieKey::BlockTransactionHashes { height } => {
+                res.extend(col::BLOCK_TRANSACTION_HASH);
+                res.extend(&height.to_le_bytes());
+            }
+            #[cfg(feature = "protocol_feature_transaction_hashes_in_state")]
+            TrieKey::AccountTransactionHash { account_id, hash } => {
+                res.extend(col::ACCOUNT_TRANSACTION_HASH);
+                res.extend(hash.as_ref());
+                res.extend(account_id.as_bytes());
```

Comment:
Feels a bit odd that the order of hash and account_id is reversed with the mathc statement/variant decl

Reply:
This is intentional to reduce depth in merkle patricia tree.